### PR TITLE
impl(compute): rework non-hermetic integration test

### LINKED ISSUE #12874: Integration test flakes due to shared instance name
The resource name is shared across test runs:

https://github.com/googleapis/google-cloud-cpp/blob/5862f092e2448c9dfb13b4db9eb205de89cb023c/google/cloud/compute/integration_tests/compute_integration_test.cc#L180C30-L180C30

https://github.com/googleapis/google-cloud-cpp/blob/5862f092e2448c9dfb13b4db9eb205de89cb023c/google/cloud/compute/integration_tests/compute_integration_test.cc#L219

### PR DESCRIPTION
**Pull request by scotthart**
fixes #12874

- - -
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/12885)

### CODE REVIEW

File: google/cloud/compute/integration_tests/compute_integration_test.cc [hunk 1]
```diff
@@ -174,42 +175,47 @@ TEST_F(ComputeIntegrationTest, VerifyUpdateSendsUpdateMaskParameter) {
...
+              [client, project_id = project_id_, network_name = network.name()](
+                  auto f) mutable -> StatusOr<compute_proto::Network> {
+                auto result = f.get();
+                if (!result.ok()) return std::move(result).status();
+                return client.GetNetwork(project_id, network_name);
+              })
+          .get();
+
+  EXPECT_STATUS_OK(patch_result);
+  if (patch_result) {
```
Comment (devbww) [inline google/cloud/compute/integration_tests/compute_integration_test.cc:214]: Q.  Why do we silently allow for failure here?

**Review by scotthart — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/12885)* status: 0 of 3 files reviewed, 1 unresolved discussion (waiting on @devbww)

___
*[`google/cloud/compute/integration_tests/compute_integration_test.cc` line 214 at r1](https://reviewable.io/reviews/googleapis/google-cloud-cpp/12885#gh-1357273139:-Ng_JNE72_l-0nm0Qn27:bzbs1zy) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/a1357859e1ebe8675a7cd7f702ce640d65f61108/google/cloud/compute/integration_tests/compute_integration_test.cc#L214)):*
<details><summary><i>Previously, devbww (Bradley White) wrote…</i></summary><blockquote>

Q.  Why do we silently allow for failure here?
</blockquote></details>

I don't follow. If `patch_result` holds an error `Status` it will fail the `EXPECT_STATUS_OK`. If `patch_result` holds a value, we then interrogate that value with the `EXPECT_THAT`.

**Review by devbww — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/12885)* status: 0 of 3 files reviewed, 1 unresolved discussion (waiting on @scotthart)

___
*[`google/cloud/compute/integration_tests/compute_integration_test.cc` line 214 at r1](https://reviewable.io/reviews/googleapis/google-cloud-cpp/12885#gh-1357273139:-Ng_S0K54EJ4GfkX7pG2:bxv6p0) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/a1357859e1ebe8675a7cd7f702ce640d65f61108/google/cloud/compute/integration_tests/compute_integration_test.cc#L214)):*
<details><summary><i>Previously, scotthart (Scott Hart) wrote…</i></summary><blockquote>

I don't follow. If `patch_result` holds an error `Status` it will fail the `EXPECT_STATUS_OK`. If `patch_result` holds a value, we then interrogate that value with the `EXPECT_THAT`.
</blockquote></details>

Sorry, I glazed over the `EXPECT_STATUS_OK`.

I take it that some form of `IsOkAndHolds` was problematic?

**Review by scotthart — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/12885)* status: 0 of 3 files reviewed, 1 unresolved discussion (waiting on @devbww)

___
*[`google/cloud/compute/integration_tests/compute_integration_test.cc` line 214 at r1](https://reviewable.io/reviews/googleapis/google-cloud-cpp/12885#gh-1357273139:-Ng_W8whAYg7caNj2-Iy:b-5v5hij) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/a1357859e1ebe8675a7cd7f702ce640d65f61108/google/cloud/compute/integration_tests/compute_integration_test.cc#L214)):*
<details><summary><i>Previously, devbww (Bradley White) wrote…</i></summary><blockquote>

Sorry, I glazed over the `EXPECT_STATUS_OK`.

I take it that some form of `IsOkAndHolds` was problematic?
</blockquote></details>

My MatcherFu was not up to the task, but I'm happy to update it (in another PR) if you have suggestions.

**Review by devbww — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/12885)* status: 0 of 3 files reviewed, 1 unresolved discussion

___
*[`google/cloud/compute/integration_tests/compute_integration_test.cc` line 214 at r1](https://reviewable.io/reviews/googleapis/google-cloud-cpp/12885#gh-1357273139:-Ng_bu8o0yFoy62iRn7Y:b-6idsnm) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/a1357859e1ebe8675a7cd7f702ce640d65f61108/google/cloud/compute/integration_tests/compute_integration_test.cc#L214)):*
<details><summary><i>Previously, scotthart (Scott Hart) wrote…</i></summary><blockquote>

My MatcherFu was not up to the task, but I'm happy to update it (in another PR) if you have suggestions.
</blockquote></details>

I would try something along the lines of:
```
  EXPECT_THAT(
      patch_result,
      IsOkAndHolds(ResultOf(
          "has routing_config.routing_mode",
          [](auto const& n) { return n.routing_config().routing_mode(); },
          Eq("GLOBAL"))));
```

**Pull request merged**